src/include/btmem.h

```diff
@@ -1165,14 +1165,15 @@ struct __wt_update {
     volatile uint8_t prepare_state; /* prepare state */
 
 /* AUTOMATIC FLAG VALUE GENERATION START 0 */
-#define WT_UPDATE_DS 0x01u                       /* Update has been written to the data store. */
-#define WT_UPDATE_FIXED_HS 0x02u                 /* Update that fixed the history store. */
-#define WT_UPDATE_HS 0x04u                       /* Update has been written to history store. */
-#define WT_UPDATE_PREPARE_RESTORED_FROM_DS 0x08u /* Prepared update restored from data store. */
-#define WT_UPDATE_RESTORED_FAST_TRUNCATE 0x10u   /* Fast truncate instantiation */
-#define WT_UPDATE_RESTORED_FROM_DS 0x20u         /* Update restored from data store. */
-#define WT_UPDATE_RESTORED_FROM_HS 0x40u         /* Update restored from history store. */
-                                                 /* AUTOMATIC FLAG VALUE GENERATION STOP 8 */
+#define WT_UPDATE_DS 0x01u                        /* Update has been written to the data store. */
+#define WT_UPDATE_FIXED_HS 0x02u                  /* Update that fixed the history store. */
+#define WT_UPDATE_HS 0x04u                        /* Update has been written to history store. */
+#define WT_UPDATE_PREPARE_RESTORED_FROM_DS 0x08u  /* Prepared update restored from data store. */
+#define WT_UPDATE_RECONCILED_PREPARE_IN_MEM 0x10u /* Prepared update reconciled, WT in-memory. */
```

Comment:
Do we really need the flag? I think we can directly add a tombstone for in memory if the rollbacked prepared update has WT_UPDATE_PREPARE_RESTORED_FROM_DS on it?

Reply:
Good point! I've removed the new flag and used `WT_UPDATE_PREPARE_RESTORED_FROM_DS` in 8122161, it just requires a few more `WT_CONN_IN_MEMORY` checks to work.